boilerplate/openshift/custom-catalog-osd-operator/current-version-getter.sh

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bash
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+OPERATOR_NAME=$(basename ${REPO_ROOT}  | sed 's/-catalog//')
+CERT_DIR="${REPO_ROOT}/service-account/rhcs-request/"
+CERT_FILE_NAME="custom-catalog-source-index-builder"
+PYXIS_ENDPOINT="https://pyxis.engineering.redhat.com"
+VERSIONS_DIR="${REPO_ROOT}/versions"
+
+
+# compare_versions accepts two semvers.
+# Its return value indicates how they compare:
+# 0: the semvers are equal
+# 1: the first semver is newer
+# 2: the second semver is newer
+compare_versions() {
+    [[ $1 == $2 ]] && return 0
+    local older=$(printf "$1\n$2\n" | sort -V | head -1)
+    [[ $older == $2 ]] && return 1
+    return 2
+}
+
+function update_latest_version() {
+  # perform the API request
+  local channel_name=${1}
+  local version_in_repo=${2}
+  local version_file=${3}
+
+  local available_operator_versions=$(curl -X GET --key ${CERT_DIR}/${CERT_FILE_NAME}.key --cert ${CERT_DIR}/${CERT_FILE_NAME}.crt \
+    "${PYXIS_ENDPOINT}/v1/operators/bundles?page_size=100&organization=redhat-operators&filter=package==${OPERATOR_NAME};in_index_img==true;channel_name==${channel_name}" | \
+    jq -r "[.data[] | {bundle_image: .bundle_path, operator_version: .version, channel_name: .channel_name, operator_name: .package }]" )
+
+    jq -c '.[]' <<< ${available_operator_versions} | while read i; do
+      local upstream_operator_version=$( jq .operator_version <<< "${i}" )
+      upstream_operator_version=$(sed -e 's/^"//' -e 's/"$//' <<< ${upstream_operator_version})
+      
+      compare_versions ${upstream_operator_version} ${version_in_repo}
+      if [[ $? == 1 ]]; then
+        echo "Found newer version upstream: ${upstream_operator_version}"
+        echo ${i} |jq > ${version_file}
+      fi
```

Comment:
Since you're in a loop here, and using `>`, if there's more than one newer version, you'll end up with just the last one in your version_file. Are you guaranteed that the versions from pyxis arrive sorted? If not, you may want to consider something like:

```suggestion
      compare_versions ${upstream_operator_version} ${version_in_repo}
      if [[ $? == 1 ]]; then
        echo "Found newer version upstream: ${upstream_operator_version}"
        echo ${i} |jq > ${version_file}
        # This will make sure we end up with the latest available version
        version_in_repo=${upstream_operator_version}
      fi
```

Reply:
I was going under the assumption that new upstream operator versions are only built once in a sprint, so there should only be one version that is greater than the current one. But I like this as it will make it safer to ensure we always end up with the latest version if this convention gets used by a different operator that release operator versions more often (than a week).